src/server/data_store/lru_cache_data_store.cc

```diff
@@ -0,0 +1,51 @@
+#include "lru_cache_data_store.h"
+
+::std::string KeyValuePair::get_key() const {
+  return key;
+}
+
+::std::string KeyValuePair::get_value() const {
+  return value;
+}
+
+// This is not thread safe and very naive implementation.
+::LruCacheDataStore::LruCacheDataStore(::std::uint32_t capacity) {
+  capacity = capacity;
+  key_iterator_map = ::std::unordered_map<::std::string, ::std::list<KeyValuePair>::iterator>();
+  pairs = ::std::list<KeyValuePair>();
+}
+
+::LruCacheDataStore::~LruCacheDataStore() {}
+
+::std::string LruCacheDataStore::Get(::std::string key) {
+  if (key_iterator_map.find(key) == key_iterator_map.end()) {
+    return nullptr;
+  }
+  ::std::list<KeyValuePair>::const_iterator existing_pair_iterator = key_iterator_map.at(key);
+  // TODO(joejoevictor): This seems dumb. I wonder if we could avoid the copy.
+  KeyValuePair pair_to_be_inserted = KeyValuePair(key, (*existing_pair_iterator).get_value());
+  pairs.erase(existing_pair_iterator);
```

Comment:
I am not sure how erasing iterator in C++ will behave.. will that leave holes in the list ? hmm

Reply:
So stl's list is just a doubly linked list, erase an iterator will erase the node pointed by the iterator. You could see the reference for erase method in http://www.cplusplus.com/reference/list/list/erase/